scripts/generateGitHubStars.js

```diff
@@ -33,6 +37,8 @@ async function readHomepage(manifestFile) {
 }
 
 async function main(vcpkgDir, destDir, githubToken) {
+    try {
```

Comment:
Why are we wrapping everything in a `try{}` block?

Reply:
we are encountering exceptions while generating the stars information.

Reply:
What is the error you are getting?

Reply:
RequestError [HttpError]: Not Found
    at D:\gitbub\vcpkg\vcpkg.github.io\node_modules\@octokit\r
equest\dist-node\index.js:88:21
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5
)
    at async getGitHubStars (D:\gitbub\vcpkg\vcpkg.github.io\s
cripts\generateGitHubStars.js:21:22)
    at async main (D:\gitbub\vcpkg\vcpkg.github.io\scripts\gen
erateGitHubStars.js:57:23) {
  status: 404,
  response: {
    url: 'https://api.github.com/repos/artem-ogre/CDT.git',
    status: 404,
    headers: {
      'access-control-allow-origin': '*',
      'access-control-expose-headers': 'ETag, Link, Location, Retry-After, X-Git
Hub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit
-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-In
terval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sun
set',
      'content-encoding': 'gzip',
      'content-security-policy': "default-src 'none'",
      'content-type': 'application/json; charset=utf-8',
      date: 'Wed, 26 Jul 2023 13:28:05 GMT',
      'github-authentication-token-expiration': '2023-10-22 12:28:33 UTC',
      'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-ori
gin',
      server: 'GitHub.com',
      'strict-transport-security': 'max-age=31536000; includeSubdomains; preload
',
      'transfer-encoding': 'chunked',
      vary: 'Accept-Encoding, Accept, X-Requested-With',
      'x-accepted-oauth-scopes': 'repo',
      'x-content-type-options': 'nosniff',
      'x-frame-options': 'deny',
      'x-github-api-version-selected': '2022-11-28',
      'x-github-media-type': 'github.v3; format=json',
      'x-github-request-id': '32B4:25D348:3F828:42230:64C11F65',
      'x-oauth-scopes': 'admin:enterprise, admin:gpg_key, admin:org, admin:org_h
ook, admin:public_key, admin:repo_hook, admin:ssh_signing_key, audit_log, codesp
ace, delete:packages, gist, notifications, project, repo, user, workflow, write:
discussion, write:packages',
      'x-ratelimit-limit': '5000',
      'x-ratelimit-remaining': '2775',
      'x-ratelimit-reset': '1690379382',
      'x-ratelimit-resource': 'core',
      'x-ratelimit-used': '2225',
      'x-xss-protection': '0'
    },
    data: {
      message: 'Not Found',
      documentation_url: 'https://docs.github.com/rest/repos/repos#get-a-repository'
    }
  },
  request: {
    method: 'GET',
    url: 'https://api.github.com/repos/artem-ogre/CDT.git',
    headers: {
      accept: 'application/vnd.github.v3+json',
      'user-agent': 'octokit-rest.js/19.0.4 octokit-core.js/4.0.5 Node.js/18.17.0 (win32; x64)',
      authorization: 'token [REDACTED]'
    },
    request: { hook: [Function: bound bound register] }
  }
}
Failed to get stars for https://github.com/kubernetes-client/c/
Not a valid GitHub repository URL.